src/services/ldap-config/errors.js

```diff
@@ -0,0 +1,29 @@
+const reqlib = require('app-root-path').require;
+const LDAPConnectionError = require('../ldap/LDAPConnectionError');
+
+const { NotAuthenticated } = reqlib('src/errors');
+
+const errorHandlers = [
+	{
+		match: (error) => error instanceof NotAuthenticated,
+		type: 'WRONG_CREDENTIALS',
+		message: () => 'Wrong search-user credentials',
+	},
+	{
+		match: (error) => error instanceof LDAPConnectionError,
+		type: 'CONNECTION_ERROR',
+		message: (error) => `${error.data.code}: ${error.data.message}`,
+	},
+	{
+		match: (error) => error.lde_message === 'No Such Object',
+		type: 'WRONG_SEARCH_PATH',
+		message: () => 'Wrong search or base path',
+	},
+	{
+		match: (error) => error.message === 'Invalid configuration object',
+		type: 'INVALID_CONFIGURATION_OBJECT',
+		message: () => 'Invalid configuration object',
```

Comment:
I wonder why for those two cases we are using string comparision and not comparing error types like above?

Reply:
it's a bit messy, since we cannot import the errors from the ldapjs library and the second one is a GeneralError, so that's not really helpful

Reply:
Ok, then it is ok.